### PR TITLE
[IMP] l10n_uy: Type of doc based on type of ID

### PR DESCRIPTION
**Pull request by mem-adhoc**
Description of the Issue/Feature Addressed by This PR:
The current system sets the default document type to e-Ticket, regardless of the partner's identification type. However, for usability purposes, if the identification type is RUT/RUC, the document type should default to e-Factura, as this is the correct choice in 95% of cases.

Current Behavior Before PR:
The default document type is always set to e-Ticket, irrespective of the identification type.

Desired Behavior After PR:
The system will automatically select the appropriate document type (either e-Factura or e-Ticket) in the account.move module, based on the partner's identification type (RUT/RUC or other). This will reduce the need for manual adjustments. Users will still have the flexibility to select other document types to account for any exceptions.

Latam-task: 1273
Adhoc-side-task: 44194

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_uy/models/account_move.py [hunk 1]
```diff
@@ -48,3 +48,16 @@ def _get_l10n_latam_documents_domain(self):
...
         return domain
+
+    def _compute_l10n_latam_document_type(self):
+        """
+        The following rules apply for determining document types based on the partner's identification:
+        RUT/RUC (Uruguay): Automatically select e-factura.
+        CI (Cedula de Identidad): Automatically select e-ticket, with the following considerations:
+        """
+        super()._compute_l10n_latam_document_type()
+        for rec in self:
```
Comment (zaoral) [inline addons/l10n_uy/models/account_move.py:59]: add and if, only apply for urugayan invoices

File: addons/l10n_uy/models/account_move.py [hunk 2]
```diff
@@ -48,3 +48,16 @@ def _get_l10n_latam_documents_domain(self):
                 allowed_codes = set(codes).intersection(set(matching_subtype_codes[0]))
                 domain += [("code", "in", tuple(allowed_codes))]
         return domain
+
+    def _compute_l10n_latam_document_type(self):
+        """
+        The following rules apply for determining document types based on the partner's identification:
+        RUT/RUC (Uruguay): Automatically select e-factura.
+        CI (Cedula de Identidad): Automatically select e-ticket, with the following considerations:
```
Comment (zaoral) [inline addons/l10n_uy/models/account_move.py:56]: " following considerations: ".. maybe there is something missing?

File: addons/l10n_uy/models/account_move.py [hunk 3]
```diff
@@ -48,3 +48,16 @@ def _get_l10n_latam_documents_domain(self):
...
+        """
+        super()._compute_l10n_latam_document_type()
+        for rec in self:
+            if rec.partner_id.l10n_latam_identification_type_id == self.env.ref('l10n_uy.it_ci'):
+                rec.l10n_latam_document_type_id = self.env.ref('l10n_uy.dc_e_ticket').id
+            if rec.partner_id.l10n_latam_identification_type_id.name == self.env.ref('l10n_uy.it_rut'):
```
Comment (zaoral) [inline addons/l10n_uy/models/account_move.py:62]:
```suggestion
            if rec.partner_id.l10n_latam_identification_type_id == self.env.ref('l10n_uy.it_rut'):
```

File: addons/l10n_uy/models/account_move.py [hunk 4]
```diff
@@ -48,3 +48,16 @@ def _get_l10n_latam_documents_domain(self):
...
+                rec.l10n_latam_document_type_id = self.env.ref('l10n_uy.dc_e_inv').id
```
Comment (zaoral) [inline addons/l10n_uy/models/account_move.py:63]: Please add unit test to ensure that create a invoice with a contact with RUC will suggest e-factura. and create a invoice with a partner with CI will automatically select the e-ticket document.

File: addons/l10n_uy/tests/test_doc_types.py [hunk 5]
```diff
@@ -61,3 +69,13 @@ def test_debit_note(self):
         self.assertEqual(debit_note.l10n_latam_document_type_id.code, "123", "Not Export e-Invoice Debit Note")
         expected_docs = ["123"] if self.env['ir.module.module']._get('l10n_uy_edi').state == 'installed' else ['123', '223']
         self.assertEqual(debit_note.l10n_latam_available_document_type_ids.mapped("code"), expected_docs, "Bad Domain")
+
+    def test_doc_type_byid(self):
```
Comment (zaoral) [inline addons/l10n_uy/tests/test_doc_types.py:73]:
```suggestion
    def test_default_doc_type_by_id(self):
```

**Review by zaoral — COMMENTED** [review]
Hi @mem-adhoc 
I send you some feedback about this PR
Best Regards

File: addons/l10n_uy/models/account_move.py [hunk 6]
```diff
@@ -48,3 +48,16 @@ def _get_l10n_latam_documents_domain(self):
...
                 domain += [("code", "in", tuple(allowed_codes))]
         return domain
+
+    def _compute_l10n_latam_document_type(self):
+        """
+        The following considerations apply for determining document types based on the partner's identification:
+        RUT/RUC (Uruguay): Automatically select e-factura.
+        Other documents (Example: CI, PAS, NIE, NIFE, etc.): Automatically select e-ticket,
+        with the following considerations:
+        """
```
Comment (fvz-odoo) [inline addons/l10n_uy/models/account_move.py:58]:
```suggestion
        with the following considerations: 
        """
```
```suggestion
        Other documents (Example: CI, PAS, NIE, NIFE, etc.): Automatically select e-ticket
        """
```

Comment (fvz-odoo) [inline addons/l10n_uy/models/account_move.py:58]:
```suggestion
        Other documents (Example: CI, PAS, NIE, NIFE, etc.): Automatically select e-ticket
        """
```

**Comment by fvz-odoo** [comment]
Hello @mem-adhoc 

A couple of things to check here: 

- I made a comment on the code, just a minor change to remove the "with the following considerations"
- The description of the PR change it to 

**Description of the Issue/Feature Addressed by This PR:**
The current system sets the default document type to e-Ticket, regardless of the partner's identification type. However, for usability purposes, if the identification type is RUT/RUC, the document type should default to e-Factura, as this is the correct choice in 95% of cases.

**Current Behavior Before PR:**
The default document type is always set to e-Ticket, irrespective of the identification type.

**Desired Behavior After PR:**
The system will automatically select the appropriate document type (either e-Factura or e-Ticket) in the account.move module, based on the partner's identification type (RUT/RUC or other). This will reduce the need for manual adjustments. Users will still have the flexibility to select other document types to account for any exceptions.

Latam-task: 1273
Adhoc-side-task: 44194


- ci/style, ci/runbot and ci/codeowner are failing, could you please check?


Other than these changes, I checked functionally and LGTM, once you correct the above points, you can go ahead and create the PR (move it from draft to a real PR).

Thanks and cheers!

Fernando

@zaoral

**Comment by mem-adhoc** [comment]
@fvz-odoo 
Thank you very much, Fer, for your suggestions. The changes are ready! Cheers.

**Comment by mem-adhoc** [comment]
We close this PR, moved the changes to enterprise here https://github.com/odoo/enterprise/pull/72305

**Pull request closed without merging**